lib/matplotlib/category.py

```diff
@@ -168,6 +170,23 @@ def __init__(self, data=None):
         if data is not None:
             self.update(data)
 
+    @staticmethod
+    def _strs_are_convertible(vals):
+        """
+        Helper method to see if list of strings can all be cast to float or
+        parsed as date.
+        """
+
+        for val in vals:
```

Comment:
I still don't see the point of this second loop, like it's still checking every variable individually, and that's the same as the ordered dict loop. If you want to check all values together, that's an `if all (self.is_convertable(val) for val in data): raise cbook.warn_external)`

Reply:
This way only does the check once if the first string in the list is “a”.  The other way does it for every element.

Reply:
I think if you're bother with the check, then it should check every unique element because that's the only situation in which a user maybe intended it not to be strings. If they're mixing string floats with string words, the advice in the warning doesn't apply anway.  

Granted, I'm leaning towards @anntzer 's thinking that instead the axis lock PR should be revisited.

Reply:
I think there is some confusion.  This is just checking if all the elements are strings that are number like, and if they are it emits a warning.   As soon as it finds a string that can’t be cast to float or a date, it stops checking and doesnt emit a warning. There is no need to check the rest of the strings at that point. 

I am not clear what this has to do w axis locking.  There is no question that the converter should be the categorical converter, the problem is users who accidentally pass a bunch of number strings to plot because   csvread has returned the numbers as strings.  Those users should cast to float before passing to plot, but they have no way to know that and get confusing results.

Reply:
Away from serious reviewing for the Christmas period but I think Jody's approach is right here.